service.rb

```diff
@@ -1,24 +1,22 @@
-require 'socket'
 require 'cgi'
 require 'uri'
 require 'yaml/store'
+require 'rack/handler/puma'
+require 'rack'
 
 require_relative './http_request'
 
 class Service
   def initialize(port)
-    server = TCPServer.new(port)
-    store = YAML::Store.new("daily_data.yml")
-
-    loop do
-      client = server.accept
-      request_line = client.readline
-      method_token, target, version_number = request_line.split
+    app = -> environment {
+      method_token = environment['REQUEST_METHOD']
+      target   = environment['PATH_INFO']
+      store = YAML::Store.new("daily_data.yml")
 
       case [method_token, target]
       when ["GET", "/show/data"]
         content_type = "text/html"
-        response_status_code = "200 OK"
+        status = "200 OK"
```

Comment:
I’m a little surprised that this works, since [the spec says](https://github.com/rack/rack/blob/master/SPEC.rdoc#label-The+Status) that `status` “must be an `Integer` greater than or equal to 100”, but this isn’t an `Integer`, it’s a `String`.

It looks like Puma is lenient: it [calls `#to_i`](https://github.com/puma/puma/blob/v5.2.0/lib/puma/request.rb#L81) on whatever `status` you return, so in this case it’s fine as long as it’s something that Ruby knows how to coerce into an `Integer`. Still, it might be clearer (and shorter!) for you to return an `Integer` directly.